lib/ecmascript.ts

```diff
@@ -3233,444 +3375,468 @@ export const ES = ObjectAssign({}, ES2020, {
-      const dateAdd = calendar.dateAdd;
-      const firstAddOptions = ObjectCreate(null);
-      const intermediate = ES.CalendarDateAdd(calendar, datePart, dateDuration1, firstAddOptions, dateAdd);
-      const secondAddOptions = ObjectCreate(null);
-      const end = ES.CalendarDateAdd(calendar, intermediate, dateDuration2, secondAddOptions, dateAdd);
-
-      const dateLargestUnit = ES.LargerOfTwoTemporalUnits('day', largestUnit);
-      const differenceOptions = ObjectCreate(null);
-      differenceOptions.largestUnit = dateLargestUnit;
-      ({ years, months, weeks, days } = ES.CalendarDateUntil(calendar, datePart, end, differenceOptions));
-      // Signs of date part and time part may not agree; balance them together
-      ({ days, hours, minutes, seconds, milliseconds, microseconds, nanoseconds } = ES.BalanceDuration(
-        days,
-        h1 + h2,
-        min1 + min2,
-        s1 + s2,
-        ms1 + ms2,
-        µs1 + µs2,
-        ns1 + ns2,
-        largestUnit
-      ));
-    } else {
-      // relativeTo is a ZonedDateTime
-      const TemporalInstant = GetIntrinsic('%Temporal.Instant%');
-      const timeZone = GetSlot(relativeTo, TIME_ZONE);
-      const calendar = GetSlot(relativeTo, CALENDAR);
-      const intermediateNs = ES.AddZonedDateTime(
-        GetSlot(relativeTo, INSTANT),
-        timeZone,
-        calendar,
-        y1,
-        mon1,
-        w1,
-        d1,
-        h1,
-        min1,
-        s1,
-        ms1,
-        µs1,
-        ns1
-      );
-      const endNs = ES.AddZonedDateTime(
-        new TemporalInstant(intermediateNs),
-        timeZone,
-        calendar,
-        y2,
-        mon2,
-        w2,
-        d2,
-        h2,
-        min2,
-        s2,
-        ms2,
-        µs2,
-        ns2
-      );
-      if (largestUnit !== 'year' && largestUnit !== 'month' && largestUnit !== 'week' && largestUnit !== 'day') {
-        // The user is only asking for a time difference, so return difference of instants.
-        years = 0;
-        months = 0;
-        weeks = 0;
-        days = 0;
-        ({ seconds, milliseconds, microseconds, nanoseconds } = ES.DifferenceInstant(
-          GetSlot(relativeTo, EPOCHNANOSECONDS),
-          endNs,
-          1,
-          'nanosecond',
-          'halfExpand'
-        ));
-        ({ hours, minutes, seconds, milliseconds, microseconds, nanoseconds } = ES.BalanceDuration(
-          0,
-          0,
-          0,
-          seconds,
-          milliseconds,
-          microseconds,
-          nanoseconds,
-          largestUnit
-        ));
-      } else {
-        ({ years, months, weeks, days, hours, minutes, seconds, milliseconds, microseconds, nanoseconds } =
-          ES.DifferenceZonedDateTime(GetSlot(relativeTo, EPOCHNANOSECONDS), endNs, timeZone, calendar, largestUnit));
-      }
-    }
+export function DifferenceZonedDateTime(ns1, ns2, timeZone, calendar, largestUnit, options?: any) {
```

Comment:
if we're going to give options a type annotation, seems like we should use the correct one.